### ISSUE TITLE
Redis cluster connection connection support in spring with jedis (primary for write other for read)

### ISSUE THREAD

**Issue by Balaji001** [spring-projects/spring-session#282]
Hi ,

can you  please support for redis cluster connection in spring with jedis and redis template. i have multiple nodes from aws elastic cache primary one for write and other two nodes for read. now how can i connect both these three nodes or is it possible to connect single point(with replica group name) so that they can transfer respective nodes if it is write request it will go for primary node or if read it goes for reading capacity nodes.

**Comment by rwinch** [comment]
This seems like a Spring Data Redis question. Have you looked into using [Redis Sentinel Support](http://docs.spring.io/spring-data/redis/docs/current/reference/html/#redis:sentinel)?

cc @christophstrobl

**Comment by christophstrobl** [comment]
@Balaji001 at this time there is no configuration option available that allows a write to master, read from slave scenario. Non of the supported drivers currently offers this feature, though there are issues open for that in both the jedis and mp911de/lettuce redis drivers. In case of high availability the Redis Sentinel support @rwinch mentioned is the way to go.

**Comment by Balaji001** [comment]
How about two redis templates with two jedis connectionfactories.


Regards
Balaji.D

**Comment by christophstrobl** [comment]
Yes, having dedicated templates for read/write is one option. You can also have a look at http://stackoverflow.com/questions/29527738/spring-data-redis-master-slave-config

**Comment by Balaji001** [comment]
Hi i am using redis template to write and read transaction and now i want close connection how can i do it or if i specify template.setEnableTransactionSupport(true); is connection will close automatically?
